[Billy026] iP

Billy026 opened the pull request:
# Bob

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://example.com)

Bob frees your mind of having to remember things you need to do. It's,

- text-based  
- easy to learn  
- **~~FAST~~ _SUPER_ FAST** to use  

All you need to do is,

1. download it from [here](https://example.com).  
1. double-click it.  
1. add your tasks.  
1. let it manage your tasks for you 😊  

And it is **FREE!**  

Features:
- ✅ Managing tasks  
- ⬜ Managing deadlines (coming soon)  
- ⬜ Reminders (coming soon)  

---
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Billy026 committed:
Add settings.json

Billy026 committed:
Changed logo to display name "Bob"

Billy026 committed:
Added initial skeletal version of greeting and goodbye

Billy026 committed:
Renamed Duke.java to Bob.java

Billy026 committed:
Implement echoing

Billy026 committed:
Customised Bob's personality

Billy026 committed:
Separated code into functions

Billy026 committed:
Add storing and displaying of commands

Billy026 committed:
Implement new classes Task and InvalidTaskOperationException

Billy026 committed:
added mark and unmark capabilities

Billy026 committed:
Added 3 types of Tasks and updated Task.java

Billy026 committed:
Updated Bob.java

Billy026 committed:
Created automated test cases

Billy026 committed:
Added error handling for incorrect formatting of inputting of Deadline and Event tasks

Billy026 committed:
Created functions for most operations in if statement of storeAndList()

Billy026 committed:
Added error handling for unrecognised commands

Billy026 committed:
Update automated tests

Billy026 committed:
Handled miscellaneous errors and updated automated tests

Billy026 committed:
Implement deletion of task

Billy026 committed:
Updated automated tests

Billy026 committed:
Rewrite classes and functions for better encapsulation

Some functions in Bob.java have similar utilities. In addition, the
functions to create the different Task subtypes are very similar.

These functions could clutter the other functions in Bob.java that
are used specifically for it only. Similar functions also cause code
duplication.

Let's,
* create a new class to house all functions that execute specific
commands inputted by the user
* combine the functions that create tasks into one function

Creating a new class for these functions with similar purposes improves
encapsulation. Combining the functions that create tasks also improves
code quality.

Billy026 committed:
Improper Java coding standards

Improper styling of code, especially comments, that do not follow
Java coding standards.

Using a personal style of styling code may cause confusion for other
future programmers updating the code.

Let's update all code and comments to follow Java coding standards.

Following Java coding standards formats the code using a generally-used
guideline, improving code readability.

Refer to this website for Java coding standards:
https://se-education.org/guides/conventions/java/intermediate
.html#layout

Billy026 committed:
Arrange classes into packages

Several classes are related to one another.

The classes related to one another should be grouped together.

Let's arrange related classes into packages.

This will improve code quality and code readability.

Billy026 committed:
Task classes: implement save feature for Tasks

Tasks were only saved locally, so running the code again resets all
tasks.

Tasks should be able to persist for an effective task list.

Let's save tasks into the hard disk in order to persist their data.

Data about tasks are now able to persist, and are specific to each
user.

Billy026 committed:
Merge branch 'branch-level-7'

Billy026 committed:
Date formats: restrict formats of dates

Invalid date formats are accepted.

Invalid date formats can cause the tasks set by users to be nonsensical.
It will also invalidate the future implementation of listing tasks with
incoming due dates.

Let's
* prevent invalid date formats from being accepted
* allow conversion of date formats

This allows dates and times in tasks to be sensible, as well as maintain
a default date format.

Billy026 committed:
Date attribute classes: fix bugs in conversion of date formats.

Some valid date formats are incorrectly displayed or caused errors.

Let's
* Fix dates to be correctly displayed
* Normalize displayed date formats
* Include days of the week to the valid date formats (e.g. Sunday)
* Limit date formats to day-before-month format

We normalize displayed date formats to streamline input from
various users. This way, we avoid potential bugs, though at the
cost of a bit of user satisfaction. We limit date formats to prevent
incorrect behaviours with subsequent functions (month-before-day
format will be implemented later).

Billy026 committed:
Bob class: Added display of tasks due today

There impact of deadlines in Deadline and Event
tasks are non-existent.

Users should be alerted if tasks with deadlines are due.

Let's implement display of tasks with deadlines that are due upon
load of tasks.

This reminds users of potential tasks that they may have forgotten
about.

Billy026 committed:
Merge branch 'branch-level-8'

Billy026 committed:
All files: group functions and classes into more classes

Some classes are very long and can be divided into multiple
classes.

Creating more classes to further group similar functions allows
higher encapsulation and code quality.

Let's
* Add classes UiManager, Parser and Storage
* Move functions that handled display from Bob.java to UiManager
* Distribute functions in TaskManager to Parser and Storage

Aside from Parser and Storage mentioned in the assignment,
UiManager works similarly to Ui and TaskManager works similarly
to TaskList.

Billy026 committed:
Merge branch 'branch-a-moreOOP'

Billy026 committed:
Exception classes: updated package name

Classes are not in main package Bob.

Grouping all files in a main package Bob allows more efficient
transfer of files.

Let's put all exception files into Bob -> Exceptions

Billy026 committed:
Managers classes: updated package name

Classes are not in main package Bob.

Grouping all files in a main package Bob allows more efficient
transfer of files.

Let's put all exception files into Bob -> managers.

Billy026 committed:
Tasks classes: updated package name

Classes are not in main package Bob.

Grouping all files in a main package Bob allows more efficient
transfer of files.

Let's put all exception files into Bob -> tasks.

Billy026 committed:
Bob class: updated package name

Class is not in main package Bob.

Grouping all files in a main package Bob allows more efficient
transfer of files.

Let's put all exception files into Bob.

Billy026 committed:
Merge branch 'branch-a-packages'

Billy026 committed:
Merge branch 'add-gradle-support'

Billy026 committed:
Gradle file: update files to support Gradle

Testing of program is not automated.

Automatic testing, which is supported by Gradle, of our program
relieves developers of the workload of manually testing each
function.

Let's
* implement Gradle into our project
* ensure all versions are consistent

Billy026 committed:
Update settings.json

Some properties are missing from settings.json to support Gradle.

Let's add these properties so that our project can use Gradle
efficiently.

Billy026 committed:
Src classes: add new line to the end of all classes

All classes did not end with a new line.

This may cause problems during compilation due to errors with
parsers.

Let's add a new line to the end of all classes.

Adding a new line allows compilers to know that it is where the
class ends.

Billy026 committed:
Abstract task classes: implement JUnit tests

Only manual testing is available for abstract task classes.

We should automate testing if possible to allow more thorough
testing with minimal effort.

Let's
* create two test classes for the two abstract task classes
* implement test functions for all non-trivial functions in each
test class

Creating test classes for each non-test class improves code quality
as it is simpler to determine where errors occurred in the event
that errors do occur.

Billy026 committed:
Task classes: implement JUnit tests

Only manual testing is available for task classes.

We should automate testing if possible to allow more thorough
testing with minimal effort.

Let's
* fix a minor bug in TaskTest
* create three test classes for the three task classes
* implement test functions for all non-trivial functions in each
test class

Creating test classes for each non-test class improves code quality
as it is simpler to determine where errors occurred in the event
that errors do occur.

Billy026 committed:
DateManager class: implement JUnit tests

Only manual testing is available for DateManager.

We should automate testing if possible to allow more thorough
testing with minimal effort.

Let's
* fix a bug with year-first dates in normaliseDateFormat function
* create test class for the DateManager
* implement test functions for all non-trivial functions

Creating test classes for each non-test class improves code quality
as it is simpler to determine where errors occurred in the event
that errors do occur.

Billy026 committed:
Other Manager classes: implement JUnit tests

Only manual testing is available for the Manager classes.

We should automate testing if possible to allow more thorough
testing with minimal effort.

Let's
* create test classes for the Manager classes
* implement test functions for all non-trivial functions

Creating test classes for each non-test class improves code quality
as it is simpler to determine where errors occurred in the event
that errors do occur.

ParseManager has not been tested to meet deadline. Will implement
later.

Billy026 committed:
Merge branch 'branch-a-junit' into branch-a-gradle

Billy026 committed:
Implement JAR support

No JAR file could be created.

Change configurations in build.gradle to allow JAR file to be
created.

Let's change configurations to allow JAR file to be created.

Billy026 committed:
Merge branch 'branch-a-jar' into branch-a-gradle

Billy026 committed:
Merge branch 'branch-a-gradle'

Billy026 committed:
Add functionality to find matching tasks in task list

There is no quick way of searching for tasks.

Users may want to find out if they need to do something specific,
instead of just seeing what tasks they have left.

Let's allow users to search for tasks containing a string.

This may also help future functions that require filtering the
task list to find matching tasks.

Billy026 committed:
Test find functionality

The find function had no automatic test functions.

Implementing test functions for the find function will allow it
to be tested more efficiently.

Let's create test functions for all new functions added to support
the find function.

Test functions are added relative to which classes the functions
were from.

Again, no test functions were added for ParseManager in order to
meet the deadline. Will be added later.

Billy026 committed:
Merge branch 'branch-level-9'

Billy026 committed:
Command classes: create a class for each valid user command

All command functions are compiled in Parser.

This makes Parser unnecessary long, which worsens code readability.

Let's
* Make InvalidDateFormatException and InvalidTaskOperationException
subclasses of InvalidCommandException.
* Create an abstract class Command.
* Create a class for each valid user command.
* Create a class with only methods for type conversions, which
are used in multiple Command classes and possible other classes

Streamlining the exceptions allows all exceptions to represent
invalid user commands, which streamlines error handling. The
different exception classes allows more details for each, which
may help in logging.

Creating a class for each valid user command separates otherwise
unrelated commands into different files, improving code
readability and so, code quality.

Huge thanks to yjunyeu for inspiring the idea to make Command
classes. https://github.com/nus-cs2103-AY2425S2/ip/pull/144/files#
diff-ffe7028257bb3f644403cb7ccad35f74d0fcbcd2b6c1c1b5819745fac55fc
956

Billy026 committed:
Manager classes: modify for new Command classes

Redistribute functions in Bob, Parser and UiManager to account for
the new Command classes.

This makes the code in each class more concise and relevant to their
purpose.

Let's
* Move scanner from Bob to UiManager
* Move executeCommand to Parser, with new name parseCommand

This narrows down the scope of functions and the scanner to only
the scopes they are relevant in. Thus, code quality improves.

Billy026 committed:
All classes: improve variable names

Some variable names were too vague for their context and purpose.

Vague variable names worsens code readability.

Let's
* change all variable names to suit their context and purpose
* change all variable names to suit JavaDoc styling

This makes all variable names consistent and sensible, thus
improving code readability, and so, code quality.

Billy026 committed:
All classes: update JavaDoc styles

JavaDoc styles are outdated for new and modified functions.

Let's update all JavaDoc styles.

Billy026 committed:
Test classes: update test functions

Test functions for some classes are outdated.

Let's update affected test functions to match the current functions.

Billy026 committed:
Merge branch 'branch-post-week2-modifications'

Billy026 committed:
Checkstyle file: add support for Checkstyle

There is no automatic checking of code for problems in JavaDoc
comments.

It takes effort and time to manually check for these problems, so
we should automate it if possible.

Let's add the Checkstyle plugin for automatic JavaDoc style
checking.

Checkstyle automates the JavaDoc styling checks, giving us
developers more time to spend on more impactful things.

Billy026 committed:
Merge branch 'branch-A-CheckStyle'

yjunyeu commented on src/main/java/Bob/managers/UiManager.java:
```diff
@@ -0,0 +1,173 @@
+                + "     |    /  ___ | | __    \n"
+                + "     |  _ \\ / _ \\| |/_ \\\n"
+                + "     | |_| | |_| |  |_| |  \n"
+                + "     |____/ \\___/|_|\\__/ \n";
```
That is a nice looking logo!

yjunyeu commented on src/main/java/Bob/tasks/Task.java:
```diff
@@ -0,0 +1,124 @@
+            default:
+                throw new IllegalArgumentException("    Invalid task type: " + taskType);
+        }
+    }
```
Should there be no identation for case clauses according to the coding standard?

yjunyeu commented on src/main/java/Bob/managers/UiManager.java:
```diff
@@ -0,0 +1,173 @@
+ */
+public class UiManager {
+    // File path to save in hard disk
+    private static String FILE_PATH = "./data/bob.txt";
```
Could this be made a constant?

Billy026 committed:
Add GUI to program

All output are shown in the terminal.

This is not user-friendly to users who are not comfortable with
using primarily CLI.

Let's
* implement GUI for the user's commands and the program's responses
* give an image to the supposed character giving responses, Bob
* convert all terminal output into output shown in dialog boxes

This provides a user-friendly interface for non-technical users,
who make up the majority of target users.

Billy026 committed:
Classes with output

Bob's responses do not suit his intended personality.

The personality shown in the different responses do not match.

Let's change all responses to reflect Bob's intended personality,
a hesitant, but nice crocodile.

This gives Bob a more consistent personality, enhancing the
chat bot.

Credit for image:
Bob - https://www.reddit.com/r/PokemonScarletViolet/comments/z442kw
/what_i_wish_fuecocos_final_evolutionskeledirge/
User - https://openclipart.org/detail/247563/simple-cartoon-person

yjunyeu commented on src/main/java/Bob/managers/UiManager.java:
```diff
@@ -0,0 +1,173 @@
+            case "delete":
+                try {
+                    if (!Character.isDigit(input[1].charAt(0))) {
+                        throw new InvalidCommandException("Please provide a valid task number.");
+                    }
+                    this.parser.deleteTask(input[1].charAt(0));
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new InvalidCommandException("Please indicate which task to delete.");
+                }
+                break;
+            case "list":
+                this.parser.listTasks();
+                break;
+            case "find":
+                this.parser.findTasks(input);
+                break;
+            case "mark":
+                try {
+                    if (!Character.isDigit(input[1].charAt(0))) {
+                        throw new InvalidCommandException("Please provide a valid task number.");
+                    }
+                    this.parser.markTask(input);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new InvalidCommandException("Please indicate which task to mark.");
+                }
+                break;
+            case "unmark":
+                try {
+                    if (!Character.isDigit(input[1].charAt(0))) {
+                        throw new InvalidCommandException("Please provide a valid task number.");
+                    }
+                    this.parser.unmarkTask(input);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    throw new InvalidCommandException("Please indicate which task to unmark.");
+                }
+                break;
+            default:
+                throw new InvalidCommandException("I don't understand.");
+        }
+    }
```
Should there be no identations for switch statements as per the coding standard?

yjunyeu reviewed (commented):
Code looks really good and neat. I really like how you made your DateManager more flexible to accept different date formats.
Some cosmetic changes suggested.

shashwatchan commented on src/main/java/Bob/tasks/Task.java:
```diff
@@ -0,0 +1,124 @@
+     * 
+     * @throws InvalidTaskOperationException when task has already been completed.
+     */
+    public void check() throws InvalidTaskOperationException {
```
Perhaps a more informative name might be better as check could have many meanings depending on the context?

shashwatchan commented on src/main/java/managers/UiManager.java:
```diff
@@ -0,0 +1,175 @@
+    /**
+     * Displays a greeting on launch of main activity.
+     */
+    private void greeting() {
```
Perhaps include "display" in the function name to make it very clear what it does?

shashwatchan commented on src/main/java/managers/DateManager.java:
```diff
@@ -0,0 +1,184 @@
+        dayMap.put("sat", 6);
+        dayMap.put("saturday", 6);
+        dayMap.put("sun", 7);
+        dayMap.put("sunday", 7);
```
Perhaps you can make this part cleaner by using enums and/or a for loop?

shashwatchan commented on src/main/java/managers/DateManager.java:
```diff
@@ -0,0 +1,184 @@
+            String temp = dateParts[0];
+            dateParts[0] = dateParts[2];
+            dateParts[2] = temp;
+
```
Perhaps you can use a built-in swap to avoid the temporary variable to make the code cleaner?

shashwatchan reviewed (commented):
I think overall your PR closely follows the naming conventions-- good job! I've added some minor comments in places where I think the code could be made clearer / cleaner.

Billy026 committed:
Rename Bob package to bob to align with Checkstyle guidelines

The ancestor package is capitalised.

This does not align with the Checkstyle guidelines.

Let's
* rename the ancestor package to bob
* refactor all import and package statements
* modify paths in the fxml and build.gradle files to reflect this
change

Ensuring that all packages are lowercase standardizes styling,
allowing more consistent code among different programmers.

Billy026 committed:
Merge branch 'branch-level-10'

Billy026 committed:
Add assertions to ensure assumptions are correct

At certain points in some functions, if one of the variables
somehow do not have their intended state, the function will create
a bug.

Although unlikely, it is still in our best interest to ensure
these variables have their intended states.

Let's:
* assert that the referenced Task in the updated Command classes
and TaskManager should not be null.
* assert that the created hashmaps and hashsets in DateManager
and Parser have the correct number of entries.

Assertions terminate the program in the case of these very unlikely
scenarios, protecting data from being corrupted.

Billy026 committed:
Replace certain for loops in TaskManager with streams

Certain functions in TaskManager use for(Task t : tasks).

These instances can be replaced with streams.

Let's replace such instances with streams, filters and forEach function
calls.

Using streams instead of for loops reduces the need for deep nesting,
which worsens code quality.

Billy026 committed:
Fix deep nesting within certain classes

Deep nesting occurs in some functions, through a combination of
for loops, while loops and if statements.

Deep nesting causes code within these nestings to be hard to track,
since they have many restrictions in different statements.

Let's ensure that there is no more than 2 nesting in all of our
functions.

Limiting deep nesting to a depth of 2 allows some flexibility with
if statements in for or while loops, but avoids confusion with any
further nesting.

Billy026 committed:
Implement SLAP

Some functions were too long and had multiple groups of
computations.

Separating these groups of computations into their own functions
helps to implement SLAP, which improves code quality.

Let's group computations with similar purposes into their own
functions.

Implementing stricter SLAP allows easy modification to more
specific parts of code, and isolates these parts for more efficient
testing.

Billy026 committed:
Avoid complicated expression

Some expressions are too complicated, and so cannot be read easily.

This makes modifying them difficult.

Let's extract boolean logic out into their own variables when
more than 3 are relevant at a time.

This prevents clustering of boolean logic, abstracting out the
details and so, making the code easier to read.

Billy026 committed:
Replace magic numbers

Many magic numbers and magic words are used throughout the classes.

Magic numbers and words can confuse readers, as they may not know
what these numbers and words mean.

Let's replace magic numbers and words with defined literals,
with explanatory variable names.

Billy026 committed:
Make the happy path prominent

Billy026 committed:
Avoid empty catch blocks

Billy026 committed:
Use the default branch

Some if statements do not have an else statement when they should
be checking if invalid conditions are fulfilled.

If we simply assume these invalid conditions will not be reached,
it will cause problems if they are reached, which will cause bugs.

Let's add else clauses to if statements and default branches to
switch cases.

Billy026 committed:
Add and modify comments for all functions

Header comments are outdated for some functions. New functions
also did not have header comments.

Up-to-date header comments are needed to maintain code readability
and styling.

Let's:
* Add header comments for all extracted new functions
* Edit header comments to suit modified functions

Billy026 committed:
Merge pull request #1 from Billy026/branch-A-CodeQuality

Enhance code quality

Billy026 committed:
Merge branch 'branch-A-Assertions'

# Conflicts:
#	src/main/java/Bob/commands/DeleteCommand.java
#	src/main/java/Bob/commands/MarkCommand.java

Billy026 committed:
Merge branch 'branch-A-Streams'

Billy026 committed:
Add time to date parts of tasks

Tasks only have date without time.

Users may want to add time to their tasks, especially for tasks
to be done in a single day.

Let's
* add time to all tasks that will be created.
* initiate time to 00:00 for all tasks without specified time.
* modify incoming tasks to account for newly added time formats.

Adding time allows more flexibility in the duration they would
like to set for their tasks with deadlines, improving user
experience.

Billy026 committed:
Add getDueDate command

There is no way to filter tasks by their deadlines.

Users may want to know which tasks they have to complete by a
specific deadline.

Let's implement getDueDate, which gets all tasks with a matching
due date as the one inputted.

Creating this command allows filtering of tasks by deadline, which
is useful if users only want to see tasks with specific deadline
from a large list of tasks.

Billy026 committed:
Merge branch 'branch-GetDuedate'

Billy026 committed:
Update Task classes test

Tests for classes in bob.tasks are outdated.

Outdated tests can miss out on incorrect computations, or wrongly
fail a test that should pass.

Let's update all tests in the bob.tasks package.

Updating tests package-by-package, instead of all at once, ensures
the bob.tasks package is thoroughly tested before moving on to the
next.

Billy026 committed:
Update Parser and Storage classes test

Tests for classes in bob.parser and bob.storage are outdated.

Outdated tests can miss out on incorrect computations, or wrongly
fail a test that should pass.

Let's update all tests in bob.parser and bob.storage packages.

Updating tests package-by-package, instead of all at once, ensures
the bob.parser and bob.storage packages are thoroughly tested
before moving on to the next.

Billy026 committed:
Update Manager classes test

Tests for classes in bob.managers are outdated.

Outdated tests can miss out on incorrect computations, or wrongly
fail a test that should pass.

Let's update all tests in the bob.managers package.

Updating tests package-by-package, instead of all at once, ensures
the bob.managers package is thoroughly tested before moving on to
the next.

Billy026 committed:
Update Command classes test

There are no test functions for classes in the bob.commands
package.

All non-trivial functions should have automated tests to catch bugs
while updating code.

Let's add test functions in the bob.commands package.

Updating tests package-by-package, instead of all at once, ensures
the bob.commands package is thoroughly tested before moving on to
the next.

Billy026 committed:
Merge branch 'branch-A-MoreTesting'

Billy026 committed:
Add restraint to createCommand for events

When creating events, the end date can be before or at the same
time as the start date.

This is unsensible, and so should be rejected, with a message
telling the user that this they may have made a mistake.

Let's
* Implement a function in DateManager to enforce this restraint
* Add test functions in DateManagerTest to test this restraint

Since we are just handling potential unwanted commands, we should
create test functions after implementing the restraint functions
to test if they work as intended.

Billy026 committed:
Add restraint to createCommand for same tasks

Tasks that already exist in the task list can be added again.

This is unsensible, and so should be rejected, with a message
telling the user that this they may have made a mistake.

Let's
* Implement a function in TaskManager to enforce this restraint
* Add test functions in TaskManagerTest to test this restraint

Since we are just handling potential unwanted commands, we should
create test functions after implementing the restraint functions
to test if they work as intended.

Billy026 committed:
Merge branch 'branch-A-MoreErrorHandling'

Billy026 committed:
Update label in user dialog box

The text in the user's DialogBox is left aligned, which may not be
user-friendly.

Since the user icon is on the right, the text should be right-aligned
to enhance user experience for the UI.

Let's change the alignment of the label to be right-aligned in
getUserDialog().

Changing the alignment in the function itself allows manual alignment
based on which character is 'speaking'.

Billy026 committed:
Merge branch 'branch-A-BetterGui'

Billy026 committed:
Create HelpCommand

Users do not have a list of all valid user commands that they can
input.

By listing all commands the user can input, it helps them utilize
the program, improving user experience.

Let's
* Create HelpCommand
* Create test functions for HelpCommand

By creating an entire command for this, it isolates this list of
commands into its own dialog box, allowing users to reference it later
on.

Billy026 committed:
Merge branch 'branch-HelpCommand'

Billy026 committed:
Update based on peer feedback

Comments on my iP branch pointed out ways I could improve code
readability.

Taking into account comments from others is important as they can
highlight points that are otherwise easy to miss out on.

Let's modify all relevant parts that were pointed out by the comments
and are feasible.

While some comments suggested changes for code readability, it would
result in massive overhauls of code with complex operations, which
would worsen code readability overall. That being said, most of the
comments were implemented and useful.

Thank you to yjunyeu and shashwatchan for pointing out parts of my
code that can be modified to improve code readability.

Billy026 committed:
Modify code based on Checkstyle

The code has JavaDoc style errors, as indicated by Checkstyle.

Resolving these Checkstyle conflicts help to maintain the JavaDoc
styling.

Let's resolve all conflicts indicated by Checkstyle.

By adhering to the JavaDoc styling, it sets a styling basis to which
other developers that may modify this code can follow. This avoids
code readability problems due to clashes in personal code styling
formats.

Billy026 committed:
Modify code based on iP code quality feedback

A few parts of the code do not follow standard code quality
practices.

As with the JavaDoc conventions, following a standardized Code
Quality convention allows multiple developers to have a
standardized styling method.

Let's modify our code based on suggestions in the iP code quality
feedback.

Maintaining proper code quality allows greater code readability.

Billy026 committed:
Fix GUI bug

GUI is unable to be displayed due to incorrect path name.

Without the GUI being displayed, users cannot enter commands and
utilize the program.

Let's update the path name in MainWindows.fxml to the correct
path name.

Doing so, allows the GUI to display, and so the program to be able
to be utilized.

Billy026 committed:
Add representative screenshot

Users cannot visualize the UI of the program without running it.

This might be something they would like to have so that they can
decide whether they want to use the program.

Let's add a representative screenshot to the docs folder.

A representative screenshot shows users some functionality the
program has and can help them differentiate this program from other
similar programs.

Billy026 committed:
Update README.md

README is outdated and doesn't go over the program's functionalities.

Users may not know how to use the program.

Let's update the README with all the valid commands a user can
input.

This way, users can cross-reference the README to learn how to
use the program.

Billy026 committed:
Update README.md

Update README to support break lines between descriptions and formatting.

The description and formatting of each command are in the same line, which may confuse the user.

Let's add a break line after each description.

This causes the formatting to be on a new line.

Billy026 committed:
Merge branch 'master' of https://github.com/Billy026/ip

Billy026 committed:
Add jar file for program

There is no way of executing the program unless it is opened from an IDE, or main is run on a terminal.

This is not user friendly, there should be a file that runs the program just from being clicked.

Let's create a jar file.

A jar file allows the program to be executed after being clicked.

Billy026 committed:
Move bob.jar.jar to releases folder

bob.jar.jar is in the main folder of the repository.

It is unintuitive where the jar file is, so users will have to search for it.

Let's move the jar file into a releases folder.

This makes it more intuitive where the jar file is, improving accessibilty.

Billy026 committed:
Merge branch 'master' of https://github.com/Billy026/ip

Billy026 committed:
Update jar file

Program could only run on Windows.

The program should be able to run as intended regardless of which OS is used.

Let's update build.gradle to handle cross-platform operations.

This way, users using any OS can enjoy the benefit of this program.

Billy026 committed:
Delete releases/bob.jar.jar

bob.jar.jar is outdated.

bob.jar.jar does not support cross-platform, which is desired.

Let's remove bob.jar.jar and replace it with a jar file that supports cross-platform operations.

This way, cross-platform is supported and it doesn't confuse users with multiple jar files.

Billy026 committed:
Update manager test classes to support cross-platform

The program can only run on Windows.

It is desirable for users to be able to run the program regardless
of the OS they use.

Let's update manager test classes to support cross-platform
functionality.

Updating classes package-by-package allows concise testing of the
individual classes, to ensure modifications do not cause bugs in
the program.

Billy026 committed:
Update parser test class to support cross-platform

The program can only run on Windows.

It is desirable for users to be able to run the program regardless
of the OS they use.

Let's update parser test class to support cross-platform
functionality.

Updating classes package-by-package allows concise testing of the
individual classes, to ensure modifications do not cause bugs in
the program.

Billy026 committed:
Update storage test class to support cross-platform

The program can only run on Windows.

It is desirable for users to be able to run the program regardless
of the OS they use.

Let's update storage test class to support cross-platform
functionality.

Updating classes package-by-package allows concise testing of the
individual classes, to ensure modifications do not cause bugs in
the program.

Billy026 committed:
Update command test classes to support cross-platform

The program can only run on Windows.

It is desirable for users to be able to run the program regardless
of the OS they use.

Let's update command test classes to support cross-platform
functionality.

Updating classes package-by-package allows concise testing of the
individual classes, to ensure modifications do not cause bugs in
the program.

Billy026 committed:
Update build.gradle and src classes to support cross-platform

The program can only run on Windows.

It is desirable for users to be able to run the program regardless
of the OS they use.

Let's update build.gradle and src classes classes to support
cross-platform functionality.

Updating classes package-by-package allows concise testing of the
individual classes, to ensure modifications do not cause bugs in
the program.

Billy026 committed:
Merge branch 'master' of https://github.com/Billy026/ip

Billy026 committed:
Update help command to be executable

The help command could not be recognised by the parser.

Users will not be able to refer to what actions they can perform if
the help command is not working as intended.

Let's
* add help to the switch statement of parser
* cap the dash length of the output of help to the maximum window
width
* add test functions to ensure help can be reached in parser test

The help command was intended to be added, but was mistakenly
left out.

Billy026 committed:
Delete releases/bob.jar

Delete bob.jar

bob.jar is outdated.

Outdated jar files do not have all the intended functionalities.

Let's delete this jar file to avoid multiple jar files in the
releases folder.

This way, users will not be confused as to which jar file to
download.

Billy026 committed:
Add new jar file

Add updated jar file.

The updated jar file has not been added.

Users will not be able to utilise the new functionalities in the
updated jar file.

Let's add this jar file to update the outdated jar file.

By deleting and adding the new jar file, users only have one jar
file to download, which avoids confusion.

Billy026 committed:
Move bob.jar into subfolder releases

jar file is in the main page of the Github repository.

Users will have to search through the main page to find the
jar file.

Let's move jar files into their own subfolders.

This way, it is intuitive where the jar files are located,
avoiding confusion among users.

Billy026 committed:
Update bob.jar

bob.jar is corrupt and cannot be processed by the application.

This corruption is causing errors during runtime and preventing
the application from functioning correctly.

Let's replace bob.jar with a new, valid file.

This approach is taken to maintain consistency with the existing
data structure while resolving the corruption issue.

Billy026 committed:
Merge branch 'master' of https://github.com/Billy026/ip

Billy026 committed:
Fix build.gradle to allow cross-platform functionality

bob.jar was not working on OS apart from Windows.

This is undersirable as it should execute regardless of the OS
used.

Let's add all dependencies for JavaFX for all 3 OS.

This way, bob.jar will run on most computers, as they use either
Windows, Mac or Linux.

Billy026 committed:
Update error message for CreateCommand

Error message for when dates are not given to create deadlines or
events does not mention time is allowed.

Users may not know they can input time, which may worsen user
experience.

Let's modify the error message to indicate time is also allowed.

Modifying the error message instead of displaying it to the user
outright will prevent information overload as soon as the GUI starts
up.

Billy026 committed:
Update bob.jar

bob.jar does not support cross-platform functionality.

This is undesirable as it should be able to execute
regardless of the OS used.

Let's replace bob.jar with the updated jar file.

This approach is taken to maintain consistency with the existing
data structure while resolving the cross-platform issue.

Billy026 committed:
Rename Ui.png.png to Ui.png

Name of UI file incorrect.

The convention for the naming of the UI is incorrect.

Let's rename the UI file.

This way it can be tracked.
